### ISSUE TITLE
Newbie question, can't get past first character.

### ISSUE THREAD

**Issue by CptanPanic** [gwatcha/reaper-keys#38]
As a vim user, I was excited to try this. I am on windows with latest reaper-keys from git, and latest reaper. I followed the install instructions, and for example if I press a 'd', The window "Reaper keys feedback" opens up, with 'd-' listed in top, and my options below. At this point I can't type anything anymore, and have to x out the window. What am I missing, what is supposed to happen? I also tried typing spc, and then h, but nothing happens. 

Thanks,

**Comment by elgiovanno** [comment]
> As a vim user, I was excited to try this. I am on windows with latest reaper-keys from git, and latest reaper. I followed the install instructions, and for example if I press a 'd', The window "Reaper keys feedback" opens up, with 'd-' listed in top, and my options below. At this point I can't type anything anymore, and have to x out the window. What am I missing, what is supposed to happen? I also tried typing spc, and then h, but nothing happens. 
> 
> Thanks,
> 

Hi! This sounds related to an issue I am having on Mac. Try this and report back if it helps you continue your command sequence: instead of closing the RK window, click somewhere outside the window, like the timeline or the tracks bar and then try to continue your sequence.

**Comment by elgiovanno** [comment]
I can't reproduce your issue because I don't have windows, but if it is like the Mac OS issue, it's caused by the RK window being focused, which halts the ability to continue your sequence. The solution for me at the moment has been to unfocus the window and continue working. A minor nuisance, but it has worked for me. Plus, I just can't imagine my workflow without RK any more 🤘🏼

**Comment by gwatcha** [comment]
Yeah I think that's the issue as well. I think in the next release I'll make it detect text input and warn the user to unfocus the window, that will probably prevent a bunch of people being confused!

**Comment by CptanPanic** [comment]
Ok that sort of works, so I will have to do this everytime?
1. Type first key, pop up opens.
2. Have to click on reaper window to regain focus, and then type another character to keep going.

I also tried having the window stay there on the side, like in the video, but after a command it kind of resizes to be only one line and no longer can see any other commands.

Also related question, typically the most used keyboard command I use is the space to start and stop. What is the reaper-keys equivalent?

**Comment by elgiovanno** [comment]
Yep, for now that seems to be the way forward.

Use enter for start and stop. In one of the recent releases, a user config file was created so that you can remap keys to your liking, however I have not played around with this yet and don't know to what extent keys can be remapped. 

The window resizes to show the available commands for the context you are in. That is as expected.

**Comment by CptanPanic** [comment]
Maybe instead of just detecting text input in window and warning users, why not just add ability for it to keep reading in input from the window inputs? Or also it could programilicy give up focus?

**Comment by elgiovanno** [comment]
> Maybe instead of just detecting text input in window and warning users, why not just add ability for it to keep reading in input from the window inputs? Or also it could programilicy give up focus?

Full disclosure: I honestly don't know how it all works under the hood, my contributions have mainly been workflow related. 

What you're suggesting certainly sounds reasonable and it would be great! But I don't have the knowledge (or skills) to comment on it's implementation. If you by chance do, @gwatcha has always been very receptive and active when reviewing pull requests! 🙌🏼

When RK first started, the window did not have this issue, but it was also a lot more primitive in functionality. However, once I got going I almost didn't even have to reference it because it just felt like vim and I know that we'll enough to get by. In my case, it changed how I work for the better, to the extent that having to unfocus the window is a minor drawback in comparison to the pros of using vim-like bindings. But I see how for a first time user it could be a turn off. If you love vim, I suggest you give it a chance and I'm sure a solution will present itself :)

**Comment by gwatcha** [comment]
Unfortunately, there is no API to give up focus :( And the window could intercept some characters, but the Reaper API is also limited there because it does not recognize modifier keys.

**Comment by gwatcha** [comment]
About the what the key binding is for start and stop, you can always press <M-x> to pull up a binding list and type in the commands you want the binding for.

**Comment by mbbmbbmm** [comment]
I stumbled across the same problem on Windows.
I have two questions:
1) Does the same behaviour occur on Linux or is this inconvenience in Windows/Mac only?
2) Could you implement an option to just not show the pop up window?
I'd be willing to just learn the shortcuts on my own from a printed cheatsheet if I'd gain Vim-like comfort and speed by doing so.
As it is it seems amazing but also kind of not usable for me personally.
Thanks!

**Comment by gwatcha** [comment]
The same behaviour occurs on Linux. If you unfocus the window after the first character, does it still steal focus? It should just be in the background after the first initial re-focus.

**Comment by gwatcha** [comment]
I also just added an option to remove the auto-complete.

**Comment by mbbmbbmm** [comment]
Will test on the weekend!
